retrieval/targetmanager.go

```diff
@@ -179,6 +179,21 @@ func (tm *TargetManager) Targets() []*Target {
 	return targets
 }
 
+// DroppedTargets returns the targets dropped during relabelling.
+func (tm *TargetManager) DroppedTargets() []*Target {
+	tm.mtx.RLock()
+	defer tm.mtx.RUnlock()
+
+	targets := []*Target{}
+	for _, ps := range tm.targetSets {
+		ps.sp.mtx.RLock()
```

Comment:
Maybe worth adding a comment that this is a different mutex, I had to read the function twice to realize this.